Lib/zlib.py

```diff
@@ -165,7 +164,7 @@ def flush(self, mode=Z_FINISH):
         if mode == Z_FINISH:
             if self._gzip:
                 last += self.GZIP_TRAILER_FORMAT.pack(
-                    self._crc32.getValue(), self._size % sys.maxint)
+                    self._crc32.getValue(), self._size % 2**32)
```

Comment:
You're right. Probably the author intended  `self._size & sys.maxint`, equivalent to the modulus operation. `&` is usually a bit quicker than `%`, but it's not an issue here.

Also, I would not like to guarantee that we're not computing 2**32 every time.

Reply:
I tweaked your patch a little so that here and elsewhere `% 2**32` becomes `& _MASK32`. Even if it isn't significant for performance, the idea we're computing 2**32 each time and doing long-division just sticks a bit in the throat.